theano/tensor/opt.py

```diff
@@ -4773,7 +4845,11 @@ def local_elemwise_sub_zeros(node):
             node.op.scalar_op.nin == 2 and
             node.op.scalar_op == scalar.sub and
             node.inputs[0] == node.inputs[1]):
-        return [T.zeros_like(node.inputs[0])]
+        res = T.zeros_like(node.inputs[0])
+        # Copy over stacktrace from previous output.
+        # Julian: Pascal, is this really necessary? Is there anyway zeros_like can ever fail?
```

Comment:
If it fails for out-of-memory reasons, there could be pathological cases where it could help hypothetically.
Given that it is easy to just add the stack trace, I would keep it.